packages/flutter/lib/src/material/button_style.dart

```diff
@@ -603,30 +603,6 @@ class ButtonStyle with Diagnosticable {
     if (a == null && b == null) {
       return null;
     }
-    return _LerpSides(a, b, t);
-  }
-}
-
-class _LerpSides implements MaterialStateProperty<BorderSide?> {
-  const _LerpSides(this.a, this.b, this.t);
-
-  final MaterialStateProperty<BorderSide?>? a;
-  final MaterialStateProperty<BorderSide?>? b;
-  final double t;
-
-  @override
-  BorderSide? resolve(Set<MaterialState> states) {
-    final BorderSide? resolvedA = a?.resolve(states);
-    final BorderSide? resolvedB = b?.resolve(states);
-    if (resolvedA == null && resolvedB == null) {
-      return null;
-    }
-    if (resolvedA == null) {
-      return BorderSide.lerp(BorderSide(width: 0, color: resolvedB!.color.withAlpha(0)), resolvedB, t);
-    }
-    if (resolvedB == null) {
-      return BorderSide.lerp(resolvedA, BorderSide(width: 0, color: resolvedA.color.withAlpha(0)), t);
-    }
-    return BorderSide.lerp(resolvedA, resolvedB, t);
+    return MaterialStateBorderSide.lerp(a, b, t);
```

Comment:
Since `MaterialStateBorderSide` is a [typedef](https://github.com/flutter/flutter/blob/08adc90589abc1ab44178c3c7bf78a6bc3a0b788/packages/flutter/lib/src/material/material_state.dart#L197) now, should we not be using `WidgetStateBorderSide` directly here?

Reply:
`MaterialStateBorderSide` and `MaterialState` are still widely used in the material library, not deprecated for the source code so I'm keeping this consistency. 

When `MaterialState` is deprecated period is over then this would be replaced in the material library.